include/z64audio.h

```diff
@@ -0,0 +1,1208 @@
+    /* 4 */ ADSR_STATE_FADE,
+    /* 5 */ ADSR_STATE_HANG,
+    /* 6 */ ADSR_STATE_DECAY,
+    /* 7 */ ADSR_STATE_RELEASE,
+    /* 8 */ ADSR_STATE_SUSTAIN
+} AdsrStatus;
+
+typedef enum {
+    /* 0 */ MEDIUM_RAM,
+    /* 1 */ MEDIUM_UNK1,
+    /* 2 */ MEDIUM_CART,
+    /* 3 */ MEDIUM_DISK_DRIVE,
+    /* 4 */ MEDIUM_UNK4,
+    /* 5 */ MEDIUM_UNK5
+} SampleMedium;
+
+typedef enum {
+    /* 0 */ CODEC_ADPCM,
+    /* 1 */ CODEC_S8,
+    /* 2 */ CODEC_S16_INMEMORY,
+    /* 3 */ CODEC_SMALL_ADPCM,
+    /* 4 */ CODEC_REVERB,
+    /* 5 */ CODEC_S16
+} SampleCodec;
+
+typedef enum {
+    /* 0 */ SEQUENCE_TABLE,
+    /* 1 */ FONT_TABLE,
+    /* 2 */ SAMPLE_TABLE
+} SampleBankTableType;
+
+typedef enum {
+    /* 0 */ CACHE_TEMPORARY,
+    /* 1 */ CACHE_PERSISTENT,
+    /* 2 */ CACHE_EITHER,
+    /* 3 */ CACHE_PERMANENT
+} AudioCacheType;
+
+typedef enum {
+    /*  0 */ OCARINA_BTN_A,
+    /*  1 */ OCARINA_BTN_C_DOWN,
+    /*  2 */ OCARINA_BTN_C_RIGHT,
+    /*  3 */ OCARINA_BTN_C_LEFT,
+    /*  4 */ OCARINA_BTN_C_UP,
+    /* -1 */ OCARINA_BTN_INVALID = 0xFF
+} OcarinaButtonIdx;
+
+typedef enum {
+    /* 0x0 */ NOTE_C4,
+    /* 0x1 */ NOTE_DFLAT4,
+    /* 0x2 */ NOTE_D4,
+    /* 0x3 */ NOTE_EFLAT4,
+    /* 0x4 */ NOTE_E4,
+    /* 0x5 */ NOTE_F4,
+    /* 0x6 */ NOTE_GFLAT4,
+    /* 0x7 */ NOTE_G4,
+    /* 0x8 */ NOTE_AFLAT4,
+    /* 0x9 */ NOTE_A4,
+    /* 0xA */ NOTE_BFLAT4,
+    /* 0xB */ NOTE_B4,
+    /* 0xC */ NOTE_C5,
+    /* 0xD */ NOTE_DFLAT5,
+    /* 0xE */ NOTE_D5,
+    /* 0xF */ NOTE_EFLAT5,
+    /* -1  */ NOTE_INVALID = 0xFF
+} OcarinaNoteIdx;
+
+typedef void (*DmaHandler)(OSPiHandle* handle, OSIoMesg* mb, s32 direction);
+
+struct Note;
+struct NotePool;
+struct SequenceChannel;
+struct SequenceLayer;
+
+typedef struct AudioListItem {
+    // A node in a circularly linked list. Each node is either a head or an item:
+    // - Items can be either detached (prev = NULL), or attached to a list.
+    //   'value' points to something of interest.
+    // - List heads are always attached; if a list is empty, its head points
+    //   to itself. 'count' contains the size of the list.
+    // If the list holds notes, 'pool' points back to the pool where it lives.
+    // Otherwise, that member is NULL.
+    /* 0x00 */ struct AudioListItem* prev;
+    /* 0x04 */ struct AudioListItem* next;
+    /* 0x08 */ union {
+                   void* value; // either Note* or SequenceLayer*
+                   s32 count;
+               } u;
+    /* 0x0C */ struct NotePool* pool;
+} AudioListItem; // size = 0x10
+
+typedef struct NotePool {
+    /* 0x00 */ AudioListItem disabled;
+    /* 0x10 */ AudioListItem decaying;
+    /* 0x20 */ AudioListItem releasing;
+    /* 0x30 */ AudioListItem active;
+} NotePool; // size = 0x40
+
+// Pitch sliding by up to one octave in the positive direction. Negative
+// direction is "supported" by setting extent to be negative. The code
+// exterpolates exponentially in the wrong direction in that case, but that
```

Comment:
Is "exterpolates" an actual word?

Reply:
Probably mispelled "extrapolates"

Reply:
Yeah, it's suppose to be "extrapolate" by the looks of it:
https://github.com/n64decomp/sm64/blob/1372ae1bb7cbedc03df366393188f4f05dcfc422/src/audio/internal.h#L141